lib/prefix.c

```diff
@@ -909,53 +909,52 @@ int str2prefix(const char *str, struct prefix *p)
 static const char *prefixevpn_ead2str(const struct prefix_evpn *p, char *str,
 				      int size)
 {
-	snprintf(str, size, "Unsupported EVPN prefix");
+	char buf[ESI_STR_LEN];
+
+	snprintf(str, size, "[%d]:[%u]:[%s]:[%d]:[%s]", p->prefix.route_type,
+		 p->prefix.ead_addr.eth_tag,
+		 esi_to_str(&p->prefix.ead_addr.esi, buf, sizeof(buf)),
+		 is_evpn_prefix_ipaddr_v4(p) ? IPV4_MAX_BITLEN
+					     : IPV6_MAX_BITLEN,
+		 inet_ntoa(p->prefix.ead_addr.ip.ipaddr_v4));
```

Comment:
`inet_ntoa` is deprecated in favor of `inet_ntop` (it isn't reentrant, so this is a hard no-go for libfrr), and we have a heavily optimized version of `inet_ntop` internally as well named `frr_inet_ntop` that runs something like 6 times faster than libc's and should be preferred

Reply:
inet_ntoa is cut-paste all over the evpn code (logs mainly). but i hear you...